apps/src/lib/kits/maker/boards/microBit/LightSensor.js

```diff
@@ -67,11 +82,39 @@ export default class LightSensor extends EventEmitter {
     this.board.mb.stopStreamingAnalogChannel(sensor_channels.lightSensor); // disable light sensor
   }
 
-  //TODO
-  getAveragedValue() {}
+  // Get the averaged value over the give range of ms, adjusted within specified range
+  // TODO: Handle if the range is outside of 50 (before first sample) or 3000 (longer than our buffer).
+  // Currently returns null if range is outside of acceptable values
+  getAveragedValue(range) {
+    if (range >= SAMPLE_RATE && range <= MAX_SENSOR_BUFFER) {
+      // Divide ms range by sample rate of sensor
+      let indicesRange = Math.ceil(range / SAMPLE_RATE);
+      let sum = 0;
+      for (let i = 0; i < indicesRange; i++) {
+        // bufferIndex points to the next spot to write, so first historical
+        // data value is at (bufferIndex - 1)
+        let index =
+          (this.buffer.length + (this.bufferIndex - (i + 1))) %
+          this.buffer.length;
+        sum += this.buffer[index];
+      }
+      return calculateValueWithinRange(
+        sum / indicesRange,
+        this.state.rangeMin,
+        this.state.rangeMax
+      );
+    } else {
+      return null;
+    }
+  }
 
   setRange(min, max) {
     this.state.rangeMin = min;
     this.state.rangeMax = max;
   }
 }
+
+function calculateValueWithinRange(value, min, max) {
```

Comment:
Can we have a comment explaining what this function does?  The function name is also a bit vague.. what exactly are we calculating?